marpa/lib/Guacamole.pm

```diff
@@ -604,7 +655,7 @@ OpKeywordMsgrcvExpr           ::= OpKeywordMsgrcv Expression OpComma Expression
 
 OpKeywordMsgsndExpr           ::= OpKeywordMsgsnd Expression OpComma Expression OpComma Expression
 
-# TODO: OpKeywordMyExpr ::= OpKeywordMy Expression
+OpKeywordMyExpr               ::= OpKeywordMy Expression
```

Comment:
I don't think this is right: this would parse `my $foo, $bar` the same as `my ($foo, $bar)` but they are not. It should be `(my $foo), $bar`. So `my` acts more like an unary operator than a list operator. 

Same applies to `state`.

Reply:
How would you write it?

Reply:
`my` can only directly precede a variable name, or a parenthesized list, so `Value` seems like the place to handle them:

```
Value ::= Variable
  | Modifier Variable
  | ParenExpr
  | Modifier ParenExpr
#...

ParenExpr ::= LParen Expression RParen
Modifier ::= OpKeywordMy | OpKeywordState | OpKeywordState
```

I'm not exactly sure what is allowed inside parens following `my` except variable names. perl seems to understand full expression syntax, but complains on most operators. Nested lists are allowed.